docs/source/ML_tutorial.rst

```diff
@@ -0,0 +1,345 @@
+
+.. code-block:: output
+
+    sequence	        A_count	B_count	C_count	D_count	E_count	F_count	G_count	H_count	I_count	...	    S_freq	T_freq	U_freq	V_freq	W_freq	X_freq	Y_freq	Z_freq	charge	length
+    0	EEETVAK	            1	    0	    0	    0	    3	    0	    0	    0	    0	    ...	    0.0	        0.0	    0.0	    0.0	    0.0	    0.0	    0.0	    0.0	    2	    7
+    1	EQEEQQQQEGHNNK	    0	    0	    0	    0	    4	    0	    1	    1	    0	    ...	    0.0	        0.0	    0.0	    0.0	    0.0	    0.0	    0.0	    0.0	    3	    14
+    2	SHGGHTVISK	    0	    0	    0	    0	    0	    0	    2	    2	    1	    ...	    0.0         0.0	    0.0	    0.0	    0.0	    0.0	    0.0	    0.0	    3	    10
+    3	SGTHNMYK	    0	    0	    0	    0	    0	    0	    1	    1	    0	    ...	    0.0	        0.0	    0.0	    0.0	    0.0	    0.0	    0.0	    0.0	    2	    8
+    4	AARPTRPDK	    2	    0	    0	    1	    0	    0	    0	    0	    0	    ...	    0.0	        0.0	    0.0	    0.0	    0.0	    0.0	    0.0	    0.0	    3	    9
+
+Now we have completed all the data preprocessing steps. We have deduced a good amount of information from the amino acid sequences
+that might have influence on the retention time in the column.
+
+Now we are good to proceed on building the machine learning model.
+
+Modelling
+---------
+
+.. code-block:: python
+
+    import seaborn as sns
+    import matplotlib.pyplot as plt
+
+    from sklearn.model_selection import StratifiedKFold
+    from xgboost import XGBRegressor
+    from sklearn.model_selection import train_test_split
+    from matplotlib import pyplot
+    from sklearn.metrics import mean_squared_error
+    from sklearn.model_selection import ShuffleSplit
+
+.. code-block:: python
+
+    test_df = df.copy()
+    test_df = test_df.drop("sequence", axis=1)
+
+Now, we create the train and test set for cross-validation of the results 
+using the ``train_test_split`` function from sklearn's model_selection module with test_size 
+size equal to 30% of the data. Also, to maintain reproducibility of the results, a random_state is also assigned.
+
+.. code-block:: python
+
+    # Splitting Test data into test and validation
+    X_train, X_test, Y_train, Y_test = train_test_split(test_df, Y1_test, test_size = 0.3, random_state = 3)
+
+We will be using the ``XGBRegressor()`` class because it is clearly a regression problem as the response variable ( retention time ) is continuous.
+
+.. code-block:: python
+
+    xg_reg = XGBRegressor(n_estimators = 300, random_state = 3, max_leaves = 5, colsample_bytree = 0.7, max_depth = 7)
+
+Fit the regressor to the training set and make predictions on the test set using the familiar .fit() and .predict() methods.
+
+.. code-block:: python
+
+    xg_reg.fit(X_train, Y_train)
+    Y_pred = xg_reg.predict(X_test)
+
+Compute the mean squared error (mse) by invoking the mean_sqaured_error function from sklearn's metrics module.
+
+.. code-block:: python
+
+    mse = np.sqrt(mean_squared_error(Y_test, Y_pred))
```

Comment:
is this the mean absolute error?

Reply:
No. It is the root mean squared error.

Reply:
ah I think we had a miscommunication in the meeting. I will change that back.